modules/get-started/pages/licensing/add-license-redpanda/kubernetes.adoc

```diff
@@ -0,0 +1,326 @@
+
+== Add a new license
+
+Redpanda supports the following ways to apply a new license:
+
+- <<secret, Use a Kubernetes Secret to store the license>>.
+- <<inline, Provide the license string inline in your Helm values or manifest file>>.
+- <<console, Use Redpanda Console to upload the license to Redpanda>>.
+
+[[secret]]
+=== Use a Kubernetes Secret
+
+You can store the license in a Kubernetes Secret and reference it in your Helm values or manifest file.
+
+. Download your license file (`redpanda.license`) and create a Kubernetes Secret:
++
+[,bash]
+----
+kubectl create secret generic redpanda-license
+--from-file=license=./redpanda.license
+--namespace <namespace>
+----
++
+This command creates a Kubernetes Secret named `redpanda-license` in the specified namespace, containing the license file.
+
+. Reference the Secret:
++
+[tabs]
+======
+Helm + Operator::
++
+--
+.`redpanda-cluster.yaml`
+[,yaml]
+----
+apiVersion: cluster.redpanda.com/v1alpha2
+kind: Redpanda
+metadata:
+  name: redpanda
+spec:
+  chartRef: {}
+  clusterSpec:
+    enterprise:
+      licenseSecretRef:
+        name: redpanda-license
+        key: license
+----
+
+```bash
+kubectl apply -f redpanda-cluster.yaml --namespace <namespace>
+```
+
+--
+Helm::
++
+--
+
+[tabs]
+====
+--values::
++
+.`redpanda-license.yaml`
+[source,yaml]
+----
+enterprise:
+  licenseSecretRef:
+    name: redpanda-license
+    key: license
+----
++
+```bash
+helm upgrade --install redpanda redpanda/redpanda --namespace <namespace> --create-namespace \
+  --values redpanda-license.yaml --reuse-values
+```
+
+--set::
++
+[,bash]
+----
+helm upgrade --install redpanda redpanda/redpanda \
+  --namespace <namespace> \
+  --create-namespace \
+  --set enterprise.licenseSecretRef.name=redpanda-license \
+  --set enterprise.licenseSecretRef.key=license
+----
+
+====
+--
+======
+
+[[inline]]
+=== Provide the license inline
+
+If you prefer to provide the license string directly, you can do so as follows:
+
+[tabs]
+======
+Helm + Operator::
++
+--
+.`redpanda-cluster.yaml`
+[,yaml]
+----
+apiVersion: cluster.redpanda.com/v1alpha2
+kind: Redpanda
+metadata:
+  name: redpanda
+spec:
+  chartRef: {}
+  clusterSpec:
+    enterprise:
+      license: <license-key>
+----
+
+```bash
+kubectl apply -f redpanda-cluster.yaml --namespace <namespace>
+```
+
+--
+Helm::
++
+--
+
+[tabs]
+====
+--values::
++
+.`redpanda-license.yaml`
+[source,yaml]
+----
+enterprise:
+  license: <license-key>
+----
++
+```bash
+helm upgrade --install redpanda redpanda/redpanda --namespace <namespace> --create-namespace \
+  --values redpanda-license.yaml --reuse-values
+```
+
+--set::
++
+[,bash]
+----
+helm upgrade --install redpanda redpanda/redpanda \
+  --namespace <namespace> \
+  --create-namespace \
+  --set enterprise.license=<license-key>
+----
+
+====
+--
+======
+
+If neither the Kubernetes Secret nor the license string is provided, Redpanda looks for the license at `/etc/redpanda/redpanda.license`.
```

Comment:
This is news/surprising to me. I was under the impression that license setting had to be user initiated.

This feels a bit out of place due to the lack of an example (I don't suggest adding one as it'll be pretty gnarly and I can't vouch for the exact behavior) but it is nice to know.

Reply:
I noticed the `linux` docs say that _`rpk`_ looks for the license in `/etc/redpanda/redpanda.license`. Should this just be removed? (I had so much hope)

Reply:
Thanks - good catch 😄